### DISCUSSION TITLE
Built-in DTLS support request

### DISCUSSION THREAD

**Discussion by amitv87** [cesanta/mongoose#2873] [Feature requests]
I have found the net_builtin and tls_builtin extremely helpful in resource crunch situations.
Addition of dtls_builtin can help in realising a very low memory footprint [webrtc implementation](https://github.com/sepfy/libpeer).

**Comment by cpq** [comment]
DTLS is messy. It is not clear when exactly it can be useful, to justify the effort.
@amitv87 could you elaborate a bit more on the use case, please?

**Reply by amitv87** [reply]
Low latency VoIP is one such use case.
Many new 4G Cat-1 bis modules (Quectel/Simcom/Fibocom etc) lack VoLTE (IMS) due to limited flash and ram resources, but provide decent data transfer rates. The MCU cores in these modules are powerful enough to do realtime audio using PCM/I2S and an external codec.
Even ESP32 with wifi + lwip + mbedtls eats up a lot of ram and flash. Instead mongoose with DTLS would be much better fit for VoIP applications.

**Reply by cpq** [reply]
Thank you, that's helpful!
Could you recommend a bunch of concrete modules we can use for the reference implementation, please?

**Reply by amitv87** [reply]
Yeah Sure!
There are many modules from Quectel (Ex800x series) for China/India region but not sure about the EU availability.
Few EU specific ones are  [EG800K](https://www.soselectronic.com/en-gb/products/quectel/eg800keulc-i03-snnsa-413275?query=eg800k) [EG800G](https://soyter.pl/en/products/quectel-eg800g-eu-lte-module-8169.html?srsltid=AfmBOoqq4EDaeb8552ii8O_lp4BWXkl2bevRpDdCp-WHaFnn1oil0w-i) [LE270](https://www.fibocom.com/en/Products/LTECat1-LE270-EU.html).

Thanks.
